In [14]:
import pandas as pd
import numpy as np
import scipy as sp

# Лабораторная работа №2
## Задание 1
## Кондрашин Тимофей 211-365 Вариант 60 (20.04.2023)

### Данные

In [15]:
data = pd.DataFrame([21.0, 15.7, 29.7, 11.2, 14.7, 13.2, 18.2, 23.0, 16.1, 13.4, 20.5, 11.2, 18.2, 15.0, 23.8, 14.8])
data.T

0     1     2     3     4     5     6     7     8     9     10    11   
0  21.0  15.7  29.7  11.2  14.7  13.2  18.2  23.0  16.1  13.4  20.5  11.2  \

     12    13    14    15  
0  18.2  15.0  23.8  14.8

In [16]:
data.describe()

0
count  16.000000
mean   17.481250
std     5.048197
min    11.200000
25%    14.375000
50%    15.900000
75%    20.625000
max    29.700000

### Всего элементов ряда

In [17]:
data_len = len(data)
data_len

16

### Выборочное среднее

$$\overline x = \frac1n \sum^n_{i=1} x_i$$

In [18]:
mean = np.mean(data)
mean

17.48125

### Выборочная дисперсия

$$s^2_x = \frac{1}{n-1}\sum^n_{i=1} (x_i - \overline x)^2$$

In [19]:
var = np.var(data)
var

0    23.891523
dtype: float64

### Среднеквадратическое отклонение

$$s = \sqrt{s^2}$$

In [20]:
std = np.std(data)
std

0    4.887896
dtype: float64

### $t$-критерий Стьюдента

$$t = \frac{\overline x - \mu}{\frac{s}{\sqrt{n}}}$$

In [21]:
gamma = 0.95
t_value = sp.stats.t.ppf(gamma, data_len-1)
t_value

1.7530503556925547

### Доверительный интервал для математического ожидания при доверительной вероятности $\gamma$

#### Z-оценка

$$z={x-{\bar {X}} \over S_{x}}$$

In [22]:
sp.stats.norm.ppf(.95)

1.6448536269514722

#### Доверительный интервал для математического ожидания через $z$ оценку

$$\left(\overline x - z \cdot \frac{s}{\sqrt{n}}, \overline x + z \cdot \frac{s}{\sqrt{n}}\right)$$

In [23]:
def confidence_interval(data, gamma):
    mean = np.mean(data)
    std = np.std(data)
    n = len(data)
    z = sp.stats.norm.ppf(.95)
    return (mean - z * std / np.sqrt(n), mean + z * std / np.sqrt(n))

confidence_interval(data, 0.95)

(0    15.471282
 dtype: float64,
 0    19.491218
 dtype: float64)

#### Доверительный интервал для математического ожидания через $t$-критерий Стьюдента

$$\left(\overline x - t \cdot \frac{s}{\sqrt{n}}, \overline x + t \cdot \frac{s}{\sqrt{n}}\right)$$

In [24]:
def confidence_interval_t(data, gamma):
    mean = np.mean(data)
    std = np.std(data)
    n = data_len
    t = t_value
    return (mean - t * std / np.sqrt(n), mean + t * std / np.sqrt(n))

confidence_interval_t(data, 0.95)

(0    15.339068
 dtype: float64,
 0    19.623432
 dtype: float64)

#### Вывод

Таким образом с вероятностью 0,95 можно утверждать, что математическое ожидание лежит в интервале (15.3; 19.6)

### Доверительный интервал для среднего квадратического отклонения для того же значения $\gamma$

$$\left(\frac{s}{\sqrt{n-1}} \cdot \frac{\sqrt{n}}{z}, \frac{s}{\sqrt{n-1}} \cdot \frac{\sqrt{n}}{z}\right)$$

In [25]:
def confidence_interval_std(data, gamma):
    std = np.std(data)
    n = len(data)
    z = 1.96
    return (std / np.sqrt(n - 1) * np.sqrt(n) / z, std / np.sqrt(n - 1) * np.sqrt(n) * z)

confidence_interval_std(data, 0.95)

(0    2.575611
 dtype: float64,
 0    9.894466
 dtype: float64)

#### Вывод

Таким образом с вероятностью 0,95 можно утверждать, что стандартное отклонение генеральной совокупности лежит в интервале (2.5, 9.8)